From: Steve Sakoman <steve@sakoman.com>
Date: Sun, 24 Jan 2010 17:33:56 +0000 (-0800)
Subject: omap: mmc-twl4030: move clock input selection prior to vcc test
X-Git-Url: http://www.sakoman.com/cgi-bin/gitweb.cgi?p=linux-omap-2.6.git;a=commitdiff_plain;h=c85371ca37c39c8840672f4b8babac16007c10b8

omap: mmc-twl4030: move clock input selection prior to vcc test

otherwise it is not executed on systems that use non-twl regulators
---

diff --git a/arch/arm/mach-omap2/hsmmc.c b/arch/arm/mach-omap2/hsmmc.c
index f4a1020..3ce7f2d 100644
--- a/arch/arm/mach-omap2/hsmmc.c
+++ b/arch/arm/mach-omap2/hsmmc.c
@@ -179,15 +179,13 @@ static void hsmmc23_before_set_reg(struct device *dev, int slot,
 	if (mmc->slots[0].remux)
 		mmc->slots[0].remux(dev, slot, power_on);
 
-	if (power_on) {
-		/* Only MMC2 supports a CLKIN */
-		if (mmc->slots[0].internal_clock) {
-			u32 reg;
+	/* Only MMC2 supports a CLKIN */
+	if (mmc->slots[0].internal_clock) {
+		u32 reg;
 
-			reg = omap_ctrl_readl(control_devconf1_offset);
-			reg |= OMAP2_MMCSDIO2ADPCLKISEL;
-			omap_ctrl_writel(reg, control_devconf1_offset);
-		}
+		reg = omap_ctrl_readl(control_devconf1_offset);
+		reg |= OMAP2_MMCSDIO2ADPCLKISEL;
+		omap_ctrl_writel(reg, control_devconf1_offset);
 	}
 }
 
